os/arch/arm/src/amebad/amebad_reboot_reason.c

```diff
@@ -68,15 +68,15 @@ void up_reboot_reason_init(void)
 {
 	/* Read the same backup register for the boot reason */
 	backup_reg = BKUP_Read(BKUP_REG1);
```

Comment:
Hi Mr.Kim, the BKUP_REG1's value cannot be retained after an HW reset, so the backup_reg will be 0 after HW reset.

Reply:
That is why you will always get 4 after HW reset.

Reply:
You mean that `BKUP_REGS1` will be initialized or cleared when HW reset?

Reply:
Yes.

Reply:
Is it retained with watchdog then?

Reply:
Only HW reset will reset its value. In other cases, its value can be retained. So yes, its value can be retained after watchdog timeout.

Reply:
OK. Is there any way to retain the value even with HW reset?
Or is there any special reason to reset the value when HW reset?

Reply:
That is how BKUP_REGx works (cleared after HW reset). Currently, we treat 0 as HW reset because retaining other value makes no sense.

Reply:
Okay. I will consider more how to handle the HW reset case then.
Thanks.